android/app/src/main/java/org/fossasia/openevent/utils/AuthUtil.java

```diff
@@ -34,23 +36,45 @@
 
     private static Authenticator authenticator;
 
-    public static void signUpUser(Context context, String email, String password, ProgressBar progressBar) {
+    public static void signUpUser(String email, String password, MutableLiveData<Integer> signUpResponse) {
         APIClient.getOpenEventAPI().signUp(User.builder().email(email).password(password).build())
                 .subscribeOn(Schedulers.io())
                 .observeOn(AndroidSchedulers.mainThread())
                 .subscribe(
-                        user -> loginUser(context, email, password, progressBar),
+                        user -> {},
                         throwable -> {
-                            showProgressBar(progressBar, false);
                             showMessage(R.string.error_in_signing_up);
                             Timber.d(throwable.toString());
+                            signUpResponse.setValue(SignUpActivityViewModel.INVALID);
                         },
                         () -> {
-                            showProgressBar(progressBar, false);
                             showMessage(R.string.signed_up_successfully);
                             Timber.d("Signed up successfully");
+                            signUpResponse.setValue(SignUpActivityViewModel.VALID);
+                        });
+    }
+
+    public static void loginUserWithoutProgressBar(String email, String password, MutableLiveData<Integer> loginResponse) {
+        APIClient.getOpenEventAPI().login(new Login(email, password))
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(
+                        response -> {
+                            //Save token & email in shared preferences
+                            SharedPreferencesUtil.putString(ConstantStrings.TOKEN, response.getAccessToken());
+                            SharedPreferencesUtil.putString(ConstantStrings.USER_EMAIL, email);
```

Comment:
Don't use Android-specific classes in ViewModels

Reply:
It is using the Application context which i think shouldn't be a problem. Any other reason why we should avoid it in ViewModels ?
I'll move it to the view though.